content/guides/additional/client.md

```diff
@@ -0,0 +1,2154 @@
+        [%audio-wav flapaudioh]
+        ::
+          [%apps %flap %audio %sfx-swooshing %wav ~]
+        :_  this
+        %-  send
+        :+  200
+          ~
+        [%audio-wav flapaudiow]
+        ::
+          [%apps %flap %audio %sfx-die %wav ~]
+        :_  this
+        %-  send
+        :+  200
+          ~
+        [%audio-wav flapaudiod]
+      ==
+    ==
+  ++  dejs-action
+    =,  dejs:format
+    |=  jon=json
+    ^-  action
+    %.  jon
+    %-  of
+    :~  [%gain (ot ~[score+ni])]
+    ==
+  --
+::
+++  on-watch
+  |=  =path
+  ^-  (quip card _this)
+  ?+    path  (on-watch:default path)
+      [%http-response *]
+    ?:  =(our src):bol
+      `this
+    (on-watch:default path)
+  ==
+::
+++  on-leave  on-leave:default
+::
+++  on-peek  on-peek:default
+::
+++  on-agent
+  |=  [=wire =sign:agent:gall]
+  ^-  (quip card _this)
+  ?+    wire  (on-agent:default wire sign)
+      [%flap ~]
+    ?+    -.sign  (on-agent:default wire sign)
+      ::
+        %fact
+      ?+    p.cage.sign  (on-agent:default wire sign)
+          %flap-update
+        =/  upd  !<(update q.cage.sign)
+        ?-    -.upd
+            %lord
+          !!
+        ==
+      ==
+      ::
+        %kick
+      :_  this
+      :~  [%pass /flap %agent [src.bol %flap] %watch /updates/out]
+      ==
+      ::
+        %watch-ack
+      ?~  p.sign
+        ((slog '%flap: Subscribe succeeded!' ~) `this)
+      ((slog '%flap: Subscribe failed!' ~) `this)
+    ==
+  ==
+::
+++  on-arvo
+|=  [=wire =sign-arvo]
+  ^-  (quip card _this)
+  ?.  ?=([%eyre %bound *] sign-arvo)
+    (on-arvo:default [wire sign-arvo])
+  ?:  accepted.sign-arvo
+    %-  (slog leaf+"/apps/flap bound successfully!" ~)
+    `this
+  %-  (slog leaf+"Binding /apps/flap failed!" ~)
+  `this
+::
+++  on-fail   on-fail:default
+--
+```
+
+Then `|install our %flap` to install the app.
+
+Now when we navigate to `localhost:8080/apps/flap`, what do we see?  The game canvas is merely an empty box.  What can we do to fix this?
+
+### Serving Correctly
+
+If we investigate the Developer Tools console in our browser, we see messages to the effect that resources are unable to be located and that cross-origin requests are blocked.  Thus we need to solve two problems:
+
+1. CORS is a security policy that allows you to load resources dynamically with prior approval from the browser and server.  The quickest workaround for our case now is to simply copy `game.js` into the `script` tag.  We can also [set up CORS origins](https://developers.urbit.org/reference/arvo/eyre/guide#managing-cors-origins) for Eyre.
+
+    The first case is trivial:  remove the `src` attribute from the `script` tag and copy all of `game.js` in between the head and tail `script` tags.
+
+    The second case solves the problem a bit more generally for development:  tell your Urbit to allow `localhost` files to be served on the appropriate port.
+
+    ```hoon {% copy=true %}
+    |cors-approve 'http://localhost:8080'
+    ```
```

Comment:
I think you shouldn't have CORS issues if you're serving it directly from the ship as you seemingly are. You should only have CORS issues if you're running the app separately. So I think this shouldn't be necessary

Reply:
It's an artifact of my path to the solution.  Added to an aside in case they encounter this.